src/promptflow/promptflow/executor/_async_nodes_scheduler.py

```diff
@@ -76,7 +99,30 @@ async def execute(
         # This is because it will always call `executor.shutdown()` when exiting the `with` block.
         # Then the event loop will wait for all tasks to be completed before raising the cancellation error.
         # See reference: https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor
-        outputs = await self._execute_with_thread_pool(executor, nodes, inputs, context)
+        try:
+            if line_timeout_sec is not None:
+                outputs = await self._execute_with_timeout(executor, nodes, inputs, context, line_timeout_sec)
+            else:
+                outputs = await self._execute_with_thread_pool(executor, nodes, inputs, context)
+        except Exception as e:
+            err_msg = "Flow execution has failed."
+            if isinstance(e, LineExecutionTimeoutError):
+                err_msg = f"Line execution timeout after {line_timeout_sec} seconds."
+                context.cancel_node_runs(err_msg)
+            running_tasks = [task for task in asyncio.all_tasks(loop) if not task.done()]
+            node_names = ",".join(
+                task.get_name()
+                for task in running_tasks
+                if task.get_name() not in PF_ASYNC_NODE_SCHEDULER_EXECUTE_TASKS
```

Comment:
It seems we need to maintain a list of running task by ourself in this class, it would be unmaintainable to exclude so many task names. Let's do it in another PR.
I.e., keep a dictionary of {node_name: task}, add when create task, and del key when the task is done. Then we can avoid scanning eventloop